src/viam/sdk/spatialmath/geometry.cpp

```diff
@@ -94,7 +111,62 @@ viam::common::v1::Geometry GeometryConfig::to_proto() const {
             return geometry_;
         }
     }
-};
+}
+void GeometryConfig::set_coordinates(GeometryConfig::coordinates coord) {
```

Comment:
Since we've created a struct for coordinates that can be accessed, it seems like a good change to `GeometryConfig` would be to replace `x_`, `y_`, and `z_` with a single `coordinates` struct. Same for the `pose` struct

Reply:
Would it be okay if the struct was public rather than private?

Reply:
The values inside the struct can (and idiomatically should) be public. The field within the GeometryConfig class should probably remain private with a getter/setter as necessary.

Reply:
another thing to consider here, once we change it to a struct will I need to update the functions in pose_proto() such as set_x(x_), which I believe are proto functions?

Reply:
I think we should be able to just do `pose.set_x(coordinates.x)` instead of `pose.set_x(x_)`

Reply:
I have made the changes of adding the struct for coordinates and pose and the code looks much neater! Thanks for the suggestion. I also made the other small fixes. Unfortunately I am running into some github problems because my local code is "behind" the main branch. For extra precaution I will wait until Monday to push the final changes!